app/controllers/users/registrations_controller.rb

```diff
@@ -3,7 +3,14 @@ class RegistrationsController < Devise::RegistrationsController
     def create
       user = User::CheckSynthetic.call(params[:user][:email])
 
-      user ? redirect_to(new_user_password_path(synthetic: user.email)) : super
+      if user
+        redirect_to(new_user_password_path(synthetic: user.email))
+      else
+        super
+        if resource.persisted?
+          NotifyMailer.new_user_registered(resource.id).deliver_later
+        end
+      end
```

Comment:
I don't think putting this much conditionals in controller is a good idea
Can we refactor it somehow?

Something like this would be much nicer IMO

```
def create
  user = User::CheckSynthetic.call(params[:user][:email])
  
  return redirect_to(...) if user

  super && NotifyMailer.new_user_registered(resource).deliver_later
```
and a policy class to use in mailer to check if resource is persisted or not
What do you think about it?